### ISSUE TITLE
Make start-of-turn saves per faction

### ISSUE THREAD

**Issue by eyalroz** [wesnoth/wesnoth#6674]
### Describe the desired feature

Currently, when playing a (single-player) scenario in which you control two factions, the start-of-turn auto-save for the second faction onwards overwrite the start-of-turn auto-save's for factions action previously on that turn.

That's unfortunate - and a player can easily forget to save the "real" start-of-turn elsewhere before getting overwritten.

It would be better if, in such cases, additional factions would get a start-of-turn save suffixed with their leader name, or race name, or faction index or some combination of those.

**Comment by Discontinuum** [comment]
Not only campaign, applicable to MP too

**Comment by doofus-01** [comment]
Wouldn't save per "side" make more sense?  I'm not as familiar with MP, but at least for campaigns, it would be conceptually simpler to make default/auto-saves be something like "side X turn Y", rather than try to figure out factions or leaders.  If there is more than one leader/faction per side, that should be up to the controlling players to figure out.

**Comment by eyalroz** [comment]
@doofus-01 : I'm not exactly sure what you mean. The whole point is that for every leader you control, you want an autosave at the beginning of their turn. But - maybe I mis-phrased.

**Comment by Discontinuum** [comment]
you need a save per turn and per each side of the turn. One side can have multiple or non leaders but that isn't related

**Comment by eyalroz** [comment]
@Discontinuum : I'm still confused. Suppose I'm in a campaign scenario with two leaders fighting two other leaders; and I get to control each of the two leaders separately (i.e. when I End Turn on one of them I get to the next, and then the enemies play, then I get back to the first leader etc.) ; I need an autosave for each of my two leaders. 

Now, are these leaders different "sides"? different "factions"?

**Comment by Discontinuum** [comment]
> Now, are these leaders different "sides"?

Yes. If you didn't have to End Turn between leaders, so both leaders are movable simultaneously, that would be two leaders of a single side.
